api/v3/Engage/Signpetition.php

```diff
@@ -193,8 +193,6 @@ function _civicrm_api3_engage_signpetition_process($params) {
     }
 
     if (!empty($activity_id)) {
-      CRM_Gpapi_Processor::updateActivityWithUTM($params, $activity_id);
-
       CRM_Gpapi_Processor::resolveCustomFields($contact_data, ['source_contact_data']);
```

Comment:
I should add a follow-up ticket to refactor the `source_contact_data` code to also be part of direct calls to `Engage.startcase` - this is basically the same thing you discovered with UTM not being set if you call `Engage.startcase` directly, except for different fields.